src/Http/Http.Results/test/ServerSentEventsResultTests.cs

```diff
@@ -0,0 +1,241 @@
+    [Fact]
+    public async Task ExecuteAsync_WithObjectData_SerializesAsJson()
+    {
+        // Arrange
+        var httpContext = GetHttpContext();
+        var testObject = new TestObject { Name = "Test", Value = 42 };
+        var events = new[] { testObject }.ToAsyncEnumerable();
+        var result = TypedResults.ServerSentEvents(events);
+
+        // Act
+        await result.ExecuteAsync(httpContext);
+
+        // Assert
+        var responseBody = Encoding.UTF8.GetString(((MemoryStream)httpContext.Response.Body).ToArray());
+        Assert.Contains(@"data: {""name"":""Test"",""value"":42}", responseBody);
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_WithSsItems_SerializesDataAsJson()
+    {
+        // Arrange
+        var httpContext = GetHttpContext();
+        var testObject = new TestObject { Name = "Test", Value = 42 };
+        var events = new[] { new SseItem<TestObject>(testObject) }.ToAsyncEnumerable();
+        var result = TypedResults.ServerSentEvents(events);
+
+        // Act
+        await result.ExecuteAsync(httpContext);
+
+        // Assert
+        var responseBody = Encoding.UTF8.GetString(((MemoryStream)httpContext.Response.Body).ToArray());
+        Assert.Contains(@"data: {""name"":""Test"",""value"":42}", responseBody);
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_WithCustomJsonOptions_UsesConfiguredOptions()
+    {
+        // Arrange
+        var services = new ServiceCollection();
+        services.AddLogging();
+        services.Configure<JsonOptions>(options =>
+        {
+            options.SerializerOptions.PropertyNamingPolicy = null; // Use PascalCase
+        });
+        var httpContext = new DefaultHttpContext
+        {
+            Response = { Body = new MemoryStream() },
+            RequestServices = services.BuildServiceProvider()
+        };
+
+        var testObject = new TestObject { Name = "Test", Value = 42 };
+        var events = new[] { testObject }.ToAsyncEnumerable();
+        var result = TypedResults.ServerSentEvents(events);
+
+        // Act
+        await result.ExecuteAsync(httpContext);
+
+        // Assert
+        var responseBody = Encoding.UTF8.GetString(((MemoryStream)httpContext.Response.Body).ToArray());
+        Assert.Contains(@"data: {""Name"":""Test"",""Value"":42}", responseBody);
+    }
+
+    [Fact]
+    public async Task ExecuteAsync_WithPolymorphicType_SerializesCorrectly()
+    {
+        // Arrange
+        var httpContext = GetHttpContext();
+        var baseClass = new DerivedTestObject { Name = "Test", Value = 42, Extra = "Additional" };
+        var events = new TestObject[] { baseClass }.ToAsyncEnumerable();
+        var result = TypedResults.ServerSentEvents(events);
+
+        // Act
+        await result.ExecuteAsync(httpContext);
+
+        // Assert
+        var responseBody = Encoding.UTF8.GetString(((MemoryStream)httpContext.Response.Body).ToArray());
+        Assert.Contains(@"""extra"":""Additional""", responseBody);
```

Comment:
nit: why isn't this asserting the full body

Reply:
Was cheating a bit since we assert on the fully body serialization elsewhere but I updated this.